# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1517785109 -3600
#      Sun Feb 04 23:58:29 2018 +0100
# Node ID 21e97b50989d44126e75fa159124dcd7f7850f94
# Parent  d0221e3675efca9539c056fd3c1ea5aa2ae1f325
Workaround the presence two background axes in printout (bug #52975).

* gl2ps-print.cc (gl2ps_renderer::draw): make sure the first backplane color is
  the same as the second.

diff --git a/libinterp/corefcn/gl2ps-print.cc b/libinterp/corefcn/gl2ps-print.cc
--- a/libinterp/corefcn/gl2ps-print.cc
+++ b/libinterp/corefcn/gl2ps-print.cc
@@ -379,16 +379,23 @@ namespace octave
               {
                 size_t n_end = include_graph.find_last_not_of (' ');
                 include_graph = include_graph.substr (n_begin,
                                                       n_end - n_begin + 1);
               }
             else
               include_graph = "foobar-inc";
 
+            // FIXME: workaround gl2ps drawing 2 background planes, the first
+            //        eventually being black and producing visual artifacts
+            const figure::properties& fprop
+              = dynamic_cast<const figure::properties&> (go.get_properties ());
+            Matrix c = fprop.get_color_rgb ();
+            glClearColor (c(0), c(1), c(2), 1);
+
             // GL2PS_SILENT was removed to allow gl2ps to print errors on stderr
             GLint ret = gl2psBeginPage ("gl2ps_renderer figure", "Octave",
                                         nullptr, gl2ps_term, gl2ps_sort,
                                         (GL2PS_BEST_ROOT
                                          | gl2ps_text
                                          | GL2PS_DRAW_BACKGROUND
                                          | GL2PS_NO_PS3_SHADING
                                          | GL2PS_USE_CURRENT_VIEWPORT),
